build error: no std::has_trivial_default_constructor

Just updated xtensor, and there's this error on build:
>/usr/local/include/xtensor/xutils.hpp:623:51: error: ‘has_trivial_default_constructor’ in namespace ‘std’ does not name a template type
[build]      using xtrivially_default_constructible = std::has_trivial_default_constructor<T>;
[build]                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build] /usr/local/include/xtensor/xutils.hpp:623:46: note: suggested alternative: ‘is_trivially_default_constructible’
[build]      using xtrivially_default_constructible = std::has_trivial_default_constructor<T>;
[build]                                               ^~~
[build]                                               is_trivially_default_constructible

--- a/include/xtensor/xutils.hpp
+++ b/include/xtensor/xutils.hpp
@@ -612,7 +612,13 @@ namespace xt
      * xtrivial_default_construct implemenation *
      ********************************************/
 
-#if !defined(__GNUG__) || defined(_LIBCPP_VERSION) || (defined(_GLIBCXX_USE_CXX11_ABI) && _GLIBCXX_USE_CXX11_ABI)
+#if defined(_GLIBCXX_USE_CXX11_ABI)
+#if _GLIBCXX_USE_CXX11_ABI || (defined(_GLIBCXX_USE_DUAL_ABI) && !_GLIBCXX_USE_DUAL_ABI)
+#define XTENSOR_GLIBCXX_USE_CXX11_ABI 1
+#endif
+#endif
+
+#if !defined(__GNUG__) || defined(_LIBCPP_VERSION) || defined(XTENSOR_GLIBCXX_USE_CXX11_ABI)
 
     template <class T>
     using xtrivially_default_constructible = std::is_trivially_default_constructible<T>;
@@ -623,6 +629,7 @@ namespace xt
     using xtrivially_default_constructible = std::has_trivial_default_constructor<T>;
 
 #endif
+#undef XTENSOR_GLIBCXX_USE_CXX11_ABI
 
     /*************************
      * conditional type cast *

